reports/models.py

```diff
@@ -216,3 +217,46 @@ class UploadLevelTotals(AbstractTotals):
 
     class Meta:
         db_table = "reports_uploadleveltotals"
+
+
+class Test(models.Model):
+    id = models.TextField(primary_key=True)
+
+    external_id = models.UUIDField(default=uuid.uuid4, editable=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    repository = models.ForeignKey(
+        "core.Repository",
+        db_column="repoid",
+        related_name="tests",
+        on_delete=models.CASCADE,
+    )
+    name = models.TextField()
+    testsuite = models.TextField()
+    env = models.TextField(default="")
+
+    class Meta:
+        db_table = "reports_test"
+
+
+class TestInstance(BaseCodecovModel):
+    test = models.ForeignKey(
+        "Test",
+        db_column="test_id",
+        related_name="testinstances",
+        on_delete=models.CASCADE,
+    )
+    duration_seconds = models.FloatField()
+    outcome = models.IntegerField()
+    report = models.ForeignKey(
+        "ReportSession",
+        db_column="upload_id",
+        related_name="testinstances",
+        on_delete=models.CASCADE,
+    )
+    failure_message = models.TextField(null=True)
+    timestamp = models.TextField()
```

Comment:
Food for thought: do we need a timezone? Would we want to save a timestamp or a datetime? Does it matter?